### PR TITLE
Correctly compute free area in 2D simulation boxes using FreeVolume

### LINKED ISSUE #1473: FreeVolume computes incorrect free area in 2D simulation boxes.
### Description

`hoomd.hpmc.compute.FreeVolume` always assumes the box is 3D. This leads to errors in the free volume computation.

### Script

```python
import hoomd

snapshot = hoomd.Snapshot()
snapshot.configuration.box = (100, 100, 0, 0, 0, 0)
snapshot.particles.N = 1
snapshot.particles.types = ['A']

sim = hoomd.Simulation(device=hoomd.device.CPU(), seed=1)
sim.create_state_from_snapshot(snapshot)

mc = hoomd.hpmc.integrate.Sphere()
mc.shape['A'] = dict(diameter=1)

free_volume = hoomd.hpmc.compute.FreeVolume(test_particle_type='A', num_samples=100000)

sim.operations.integrator = mc
sim.operations.computes.append(free_volume)

sim.run(1)

print(free_volume.free_volume)
```

### Input files

_No response_

### Output

```shell
0.0
```

### Expected output

Approximately `9996.85840734641` (100*100 - pi)

### Platform

CPU, GPU, Linux, macOS

### Installation method

Compiled from source

### HOOMD-blue version

3.8.1

### Python version

3.11.1

### PR DESCRIPTION
**Pull request by melodyyzh**
## Description
Passed the system dimensionality to `getVolume` in `ComputeFreeVolume.h` and generated 0 z coordinates when the system dimensionality is 2. These changes will compute correct free area for 2D boxes. 

## Motivation and context

Resolves #1473 

## How has this been tested?
Add a 2D test to `test_compute_free_volume.py`

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: hoomd/hpmc/ComputeFreeVolumeGPU.cuh [hunk 1]
```diff
@@ -274,6 +274,11 @@ __global__ void gpu_hpmc_free_volume_kernel(unsigned int n_sample,
         Scalar yrand = hoomd::detail::generate_canonical<Scalar>(rng);
         Scalar zrand = hoomd::detail::generate_canonical<Scalar>(rng);
 
+        if (this->m_sysdef->getNDimensions() == 2)
```
Comment (joaander) [inline hoomd/hpmc/ComputeFreeVolumeGPU.cuh]:
```suggestion
        if (dim == 2)
```
`m_sydef` is not available in the GPU kernel. The caller provides the number of dimensions in the `dim` argument:
https://github.com/glotzerlab/hoomd-blue/blob/73f186e074373f62008179d7b859750c66250108/hoomd/hpmc/ComputeFreeVolumeGPU.h#L250-L278

**Review by joaander — COMMENTED** [review]
The code changes look good. I made one suggestion to fix the GPU compilation error. Now all this needs is a unit test and your name in `credits.rst`.

**Comment by melodyyzh** [comment]
pre-commit.ci autofix

**Comment by melodyyzh** [comment]
pre-commit.ci autofix

**Comment by melodyyzh** [comment]
pre-commit.ci autofix

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**